libavfilter/qsvvpp.c

```diff
@@ -874,5 +885,7 @@ int ff_qsvvpp_filter_frame(QSVVPPContext *s, AVFilterLink *inlink, AVFrame *picr
         }
     } while(ret == MFX_ERR_MORE_SURFACE);
 
+    s->last_in_pts = in_frame->frame->pts;
```

Comment:
how to handle new segments

Reply:
Sorry I didn't catch your question, could you please describe your question in detail?

Reply:
for some clips, the timestamp may not continue. not sure how to handle this in filter.

Reply:
Did you mean the input frame doesn't have a valid pts? We don't handle this case because the output and input frames have the same pts setting except deinterlacing.